### PR TITLE
feat: [ANDROAPP-3111] Add enrollment status filter to search screen

### PR DESCRIPTION
**Pull request by Balcan**
## Description
[ jira issue ](https://jira.dhis2.org/browse/ANDROAPP-3111)

## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [x] Smartphone
- [x] Tablet
## Which Android versions did you test this issue?
- [x] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [x] 8.X
- [ ] 9.X - 10.X
- [ ] Other
## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/main/res/layout/filter_org_unit.xml [hunk 1]
```diff
@@ -110,7 +109,7 @@
         <androidx.constraintlayout.widget.ConstraintLayout
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
-            android:background="?colorPrimaryDark"
+            android:background="#32000000"
```
Comment (ferdyrod) [inline app/src/main/res/layout/filter_org_unit.xml:112]: Can we set this color in the colors.xml file. I see it is set on different xml files.

File: app/src/main/res/layout/filter_period.xml [hunk 2]
```diff
@@ -20,7 +20,7 @@
     <androidx.constraintlayout.widget.ConstraintLayout
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:background="?colorPrimaryDark"
+        android:background="#32000000"
```
Comment (ferdyrod) [inline app/src/main/res/layout/filter_period.xml:23]: better to use colors.xml

File: app/src/main/res/layout/filter_state.xml [hunk 3]
```diff
@@ -20,7 +20,7 @@
     <LinearLayout
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:background="?colorPrimaryDark"
+        android:background="#32000000"
```
Comment (ferdyrod) [inline app/src/main/res/layout/filter_state.xml:23]: here too

File: app/src/main/res/layout/filter_status.xml [hunk 4]
```diff
@@ -20,7 +20,7 @@
     <LinearLayout
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:background="?colorPrimaryDark"
+        android:background="#32000000"
```
Comment (ferdyrod) [inline app/src/main/res/layout/filter_status.xml:23]: also here

File: app/src/main/java/org/dhis2/Bindings/TrackedEntityInstanceExtensions.kt [hunk 5]
```diff
@@ -25,3 +26,28 @@ fun MutableList<TrackedEntityInstance>.filterDeletedEnrollment(
...
+                    .byStatus().`in`(enrollmentStatuses)
+                    .blockingIsEmpty()
+            if (!hasEnrollmentInProgramWithStatus) {
+                iterator.remove()
+            }
+        }
+    }
+
+    return this
+}
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/Bindings/TrackedEntityInstanceExtensions.kt:53]: I miss unit tests :dancers: . Also, I'd say careful with having statics in memory :+1: . This more than a "utils" method is a use case.

Comment (Balcan) [reply app/src/main/java/org/dhis2/Bindings/TrackedEntityInstanceExtensions.kt:53]: Unit test done!

**Pull request merged**